app/auth/Authenticated.scala

```diff
@@ -35,7 +37,16 @@ trait Authenticated {
 
   val auth: AuthConnector
 
-  def authenticated(f: => AuthenticationResult => Future[Result])(implicit hc: HeaderCarrier): Future[Result] = {
+  def authenticated(f: => Future[Result])(implicit hc: HeaderCarrier): Future[Result] =
+    auth.getCurrentAuthority().flatMap { authority =>
+      Logger.debug(s"Got authority = $authority")
+      mapToAuthResult(authority) match {
+        case NotLoggedIn => Unauthorized
+        case _ => f
+      }
+    }
+
+  def authenticatedCustom(f: => AuthenticationResult => Future[Result])(implicit hc: HeaderCarrier): Future[Result] = {
```

Comment:
What does authenticaatedCustom mean? Is there a better name we can find for this?